system/Test/bootstrap.php

```diff
@@ -7,6 +7,18 @@
 // Make sure it recognizes that we're testing.
 $_SERVER['CI_ENVIRONMENT'] = 'testing';
 define('ENVIRONMENT', 'testing');
+defined('CI_DEBUG') || define('CI_DEBUG', true);
+
+// Often these constants are pre-defined, but query the current directory structure as a fallback
+defined('HOMEPATH') || define('HOMEPATH', realpath(rtrim(getcwd(), '\\/ ')) . DIRECTORY_SEPARATOR);
+$source = is_dir(HOMEPATH . 'app') 
+       ? HOMEPATH 
+       : (is_dir('vendor/codeigniter4/framework/') 
+               ? 'vendor/codeigniter4/framework/' 
+               : 'vendor/codeigniter4/codeigniter4/');
+defined('CONFIGPATH') || define('CONFIGPATH', realpath($source . 'app/Config') . DIRECTORY_SEPARATOR);
+defined('PUBLICPATH') || define('PUBLICPATH', realpath($source . 'app/Config') . DIRECTORY_SEPARATOR);
```

Comment:
```suggestion
defined('PUBLICPATH') || define('PUBLICPATH', realpath($source . 'public') . DIRECTORY_SEPARATOR);
```

Reply:
Wooooowww glad you caught that! The module I tested with must not have used PUBLICPATH at all. Copy-paste fail. 🤦‍♂️